Packs/VersaDirector/Integrations/VersaDirector/VersaDirector.py

```diff
@@ -0,0 +1,3800 @@
+    )
+    return command_results
+
+
+def obtain_auth_token(client: Client, args: Dict[str, Any]) -> CommandResults:
+    username, password = client._auth
+    client_id = args.get("client_id") or client.client_id_param
+    client_secret = args.get("client_secret") or client.client_secret_param
+
+    if not client_id or not client_secret:
+        raise DemistoException(
+            message="Auth process could not start, missing client_id and client_secret command arguments or integration parameters."
+        )
+
+    try:
+        response = client.access_token_request(username, password, client_id, client_secret)
+    except DemistoException as e:
+        if e.res.status_code == 401:
+            raise DemistoException(message="Auth process failed. Invalid access token passed in the request.")
+        else:
+            raise DemistoException(message="Auth process failed. Please check client_id and client_secret validity.")
+
+    auth_token = response.get("access_token")
+    response["client_id"] = client_id
+    response["client_secret"] = client_secret
+    demisto.setIntegrationContext(response)
+
+    command_results = CommandResults(
+        outputs_prefix="VersaDirector.Auth",
+        outputs_key_field="",
+        outputs=response,
+        raw_response=response,
+        readable_output=f"Auth request was successful.\nPlease copy Auth Token as input in the instance configuration along with the client_id and client_secret\n\nAuth Token:\n{auth_token}\n",
+    )
+    return command_results
+
+
+def appliance_list_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    page = arg_to_number(args.get("page"))
+    page_size = arg_to_number(args.get("page_size"))
+    limit = arg_to_number(args.get("limit"))
+    offset = set_offset(page, page_size)
+
+    check_limit(limit)
+
+    response = client.appliance_list_request(offset, limit)
+
+    command_results = CommandResults(
+        outputs_prefix="VersaDirector.Appliance",
+        outputs_key_field="",
+        outputs=response,
+        raw_response=response,
+        readable_output=tableToMarkdown(
+            name="Appliances",
+            t=response.get("appliance-list"),
+            headers=["name", "uuid", "ipAddress", "appType", "branchId"],
+            headerTransform=pascalToSpace,
+        ),
+    )
+    return command_results
+
+
+def organization_list_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    page = arg_to_number(args.get("page"))
+    page_size = arg_to_number(args.get("page_size"))
+    limit = arg_to_number(args.get("limit"))
+    offset = set_offset(page, page_size)
+
+    check_limit(limit)
+
+    response = client.organization_list_request(offset, limit)
+
+    # extract 'applianceuuid' list from 'appliances' dictionaty
+    appliances = []
+    for appliance in response[0].get("appliances"):
```

Comment:
Can the response be empty? response[0] might raise exception. in order to feel safe with it, we need to check that response is not an empty list before.